### PR TITLE
ANA-1315: Refactoring code

### PR DESCRIPTION
**Pull request by njmfinbourne**
# Pull Request Checklist

- [x] Read the [contributing guidelines](../blob/master/docs/CONTRIBUTING.md)
- [x] Tests pass
- [x] Raised the PR against the `develop` branch

# Description of the PR

Describe the code changes for the reviewers, explain the solution you have provided and how it fixes the issue

### CODE REVIEW

File: sdk/Lusid.Sdk.Tests/Utilities/TestDataUtilities.cs [hunk 1]
```diff
@@ -491,19 +411,19 @@ private static string GetMarketAsset(LusidInstrument option, MarketQuote.QuoteTy
...
                 FxOption fxoption = (FxOption) option;
                 marketAsset = $"{fxoption.DomCcy}/{fxoption.FgnCcy}/" + (volType == MarketQuote.QuoteTypeEnum.NormalVol ? "N" : "LN");
             }
-            
+
             if (option.InstrumentType == LusidInstrument.InstrumentTypeEnum.InterestRateSwaption)
             {
                 InterestRateSwaption swaption = (InterestRateSwaption) option;
-                var ccy = "USD"; // Note: due to swagger limitation, it is not easy to extract the domestic currency from the swaption currently. 
+                var ccy = "USD"; // Note: due to swagger limitation, it is not easy to extract the domestic currency from the swaption currently.
```
Comment (njmfinbourne) [inline sdk/Lusid.Sdk.Tests/Utilities/TestDataUtilities.cs:426]: Comment from other MR to remove this comment as it's not useful

Comment (njmfinbourne) [reply sdk/Lusid.Sdk.Tests/Utilities/TestDataUtilities.cs]: Made change.

File: sdk/Lusid.Sdk.Tests/Utilities/DemoInstrumentBase.cs [hunk 2]
```diff
@@ -119,12 +119,16 @@ internal void CallLusidGetPortfolioCashFlowsEndpoint(LusidInstrument instrument,
                 // SimpleStatic pricing is lookup pricing. As such, we upsert a quote.
                 // Note that inside CreatePortfolioAndInstrument, the method TestDataUtilities.BuildInstrumentUpsertRequest books the instrument using "ClientInternal".
                 // Hence upsert a quote using ClientInternal as the instrumentIdType.
-                var quoteRequest = TestDataUtilities.BuildQuoteRequest(
+                var quoteRequest = new Dictionary<string, UpsertQuoteRequest>();
+                TestDataUtilities.BuildQuoteRequest(
+                    quoteRequest,
+                    "req",
```
Comment (rizwansaeed) [inline sdk/Lusid.Sdk.Tests/Utilities/DemoInstrumentBase.cs:125]: what is "req"?

File: sdk/Lusid.Sdk.Tests/Utilities/InstrumentExamples.cs [hunk 3]
```diff
@@ -15,7 +15,7 @@ public static LusidInstrument GetExampleInstrument(string instrumentName)
                 nameof(Bond) => CreateExampleBond(),
                 nameof(FxForward) => CreateExampleFxForward(),
                 nameof(FxOption) => CreateExampleFxOption(),
-                nameof(InterestRateSwap) => CreateExampleInterestRateSwap(),
+                nameof(InterestRateSwap) => CreateExampleInterestRateSwap("Vanilla"),
```
Comment (rizwansaeed) [inline sdk/Lusid.Sdk.Tests/Utilities/InstrumentExamples.cs:18]: what are the other valid options? can we add a comment pointing to where they can find the list (no need to list inline)

File: sdk/Lusid.Sdk.Tests/tutorials/Ibor/StructuredResultStore.cs [hunk 4]
```diff
@@ -628,21 +628,28 @@ public void TestFindOrCalculate_PortfolioResult()
...
+                "req1",
                 luids[0],
                 QuoteSeriesId.InstrumentIdTypeEnum.LusidInstrumentId,
                 123m,
                 "USD",
-                TestDataUtilities.EffectiveAt);
-            _quotesApi.UpsertQuotes(scope, quoteRequest1);
+                TestDataUtilities.EffectiveAt,
+                QuoteSeriesId.QuoteTypeEnum.Price);
+            //_quotesApi.UpsertQuotes(scope, quoteRequest);
```
Comment (rizwansaeed) [inline sdk/Lusid.Sdk.Tests/tutorials/Ibor/StructuredResultStore.cs:641]: is this intended to be commented out?

**Review by rizwansaeed — CHANGES_REQUESTED** [review]

File: sdk/Lusid.Sdk.Tests/Utilities/DemoInstrumentBase.cs (hunk 2 above)
Comment (njmfinbourne) [reply sdk/Lusid.Sdk.Tests/Utilities/DemoInstrumentBase.cs:125]: I'm not sure.... this was used previously - I have just refactored the code but not changed this parameter. Can put a clearer name.

File: sdk/Lusid.Sdk.Tests/Utilities/InstrumentExamples.cs (hunk 3 above)
Comment (njmfinbourne) [reply sdk/Lusid.Sdk.Tests/Utilities/InstrumentExamples.cs:18]: Yes will do.

File: sdk/Lusid.Sdk.Tests/tutorials/Ibor/StructuredResultStore.cs (hunk 4 above)
Comment (njmfinbourne) [reply sdk/Lusid.Sdk.Tests/tutorials/Ibor/StructuredResultStore.cs:641]: Supposed to be removed - will do.

**Review by rizwansaeed — APPROVED** [review]

**Pull request merged**